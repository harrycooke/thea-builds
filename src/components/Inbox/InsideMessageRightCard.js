import React from 'react';
import axios from 'axios';
import {
  Grid,
  Card,
  Typography,
  withStyles,
  Modal,
  Fab,
  MobileStepper,
  Button,
  Divider,
  CircularProgress
} from "@material-ui/core";
import CloseIcon from "@material-ui/icons/Close";
import HighlightOff from "@material-ui/icons/HighlightOff";
import { makeStyles } from '@material-ui/core/styles';
import GridList from '@material-ui/core/GridList';
import GridListTile from '@material-ui/core/GridListTile';
import GridListTileBar from '@material-ui/core/GridListTileBar';
import IconButton from '@material-ui/core/IconButton';
import classnames from "classnames";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import AttachCard from "./AttachCard";
import { Document, Page, pdfjs } from "react-pdf";
import DiagnosisIcon from "../../assets/icons/DiagnosisIcon.svg";
import { isEmpty, getFileNameFromUrl } from '../../utils/functions';
import DropzonePlugin from "../NewCase/DropzonePlugin";
import { uploadFile, removeFileFromS3 } from '../../utils/s3';
import { getServerUrl } from '../../utils/functions';
import { modifyCase, setCurrentCase } from "../../actions/caseAction";
import Thumbnail from '../SharedComponents/Thumbnail';
import MomentUtils from "@date-io/moment";
import moment from "moment";

const styles = theme => ({
  rightCard: {
    overflow: "auto",
    borderRadius: 0,
    height: "92vh",
    padding: theme.spacing(4)
  },
  rightCardCollapsed: {
    width: theme.spacing(11),
    float: "right",
    justifyContent: "center",
    alignItems: "center",
    padding: 0,
    display: "flex",
    backgroundColor: "#f2f2f2"
  },
  caseInfoText: {
    [theme.breakpoints.down('sm')]: {
      fontSize: 16,
    },
    [theme.breakpoints.up('sm') ]: {
      fontSize: 20,
    },
    color: "#858585",
    marginTop: 10,
  },
  caseInfoDetailsText: {
    [theme.breakpoints.down('sm')]: {
      fontSize: 12,
    },
    [theme.breakpoints.up('sm') ]: {
      fontSize: 16,
    },
    color: "#000000",
    marginTop: 10,
  },
  additionInfoText: {
    fontSize: 20,
    color: "#848484",
    opacity: 0.4,
    marginLeft: 50
  },
  nameText: {
    fontWeight: "600",
    [theme.breakpoints.down('sm')]: {
      fontSize: 20,
    },
    [theme.breakpoints.up('sm') ]: {
      fontSize: 24,
    },
    fontSize: 24,
    color: "#454545",
    align: "center"
  },
  modalContainer: {
    minWidth: 350,
    minHeight: 350,
    width: "42%",
    height: "50%",
    outline: "none",
    alignItems: "center",
    justifyContent: "center",
    display: "flex",
    flexDirection: "column",
    position: "absolute",
    top: "27%",
  },
  modalCard: {
    width: "100%",
    height: "100%",
    outline: "none",
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    padding: 5,
  },
  img: {
    height: '100%',
    overflow: 'hidden',
    display: 'block',
    width: 'auto',
    justifyContent: "center",
    alignItems: "center",
  },
pdfView:{
    [theme.breakpoints.down('sm')]: {
        width: "69%",
    },
    [theme.breakpoints.up('sm')]: {
        width: "100%",
    }, 
    [theme.breakpoints.up('md')]: {
        width: "100%",
    },          
    overflow:'scroll',
    height:"100%",
    width: "100%",
    justifyContent: "center",
    alignItems: "center",
  },
  closeButton: {
    width: '100%',
    height: 38,
    backgroundColor: "#fff",
    marginLeft: "10%",
    paddingLeft: 15,
    paddingRight: 15,
    textTransform: "unset",
    fontSize: 18
  },
  uploadButton: {
    width: '100%',
    height: 38,
    textTransform: "unset",
    [theme.breakpoints.down('sm')]: {
      height: 35,
      fontSize: 15,
    },
    [theme.breakpoints.up('sm') ]: {
      height: 40,
      fontSize: 16,
    },
  },
})

export class InsideMessageRightCard extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentCase: '',
      openUploadFileModal:false,
      files: [],
      fileUrls: [],
      fileName: '',
      pageNumber: 1,
      isLoading: false
    }
    this.removeFile = this.removeFile.bind(this);
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    this.setState({
      currentCase: nextProps.currentCase
    })
  }

  __renderViewer = (files) => {
    let viewers = [];
    for (let i = 0; i < files.length; i += 2) {
      viewers.push(
        <Grid container justifycontent={"center"} spacing={3} key={i}>
          {files[i] !== undefined && (
            <Grid item xs={12} sm={6}>
              <AttachCard
                attachName={getFileNameFromUrl(files[i].name)}
                fileObj={files[i]}
                fileIndex={i}
                actionIcon={
                  <IconButton >
                    <HighlightOff style={{ marginLeft: 5, color:"white" }}/>
                  </IconButton>
                }
              />
            </Grid>
          )}
          {files[i + 1] !== undefined && (
            <Grid item xs={12} sm={6}>
              <AttachCard
                attachName={getFileNameFromUrl(files[i + 1].name)}
                fileObj={files[i + 1]}
                fileIndex={i + 1}
              />
            </Grid>
          )}
        </Grid>
      )
    }
    return viewers
  }

  handleFileUpload = async newFile => {
    this.setState({ isLoading: true });
    if (newFile && newFile.length) {
      for (let i = 0; i < newFile.length; i++) {
        const [promiseFile, promiseUrl, promiseName] = await uploadFile(newFile[i]); 
        this.setState({ files: [...this.state.files, promiseFile] });
        this.setState({ fileUrls: [...this.state.fileUrls, promiseUrl] });
        this.setState({ fileName: [...this.state.fileName, promiseName] });
      }
      this.setState({ isLoading: false })
    }
  }

  uploadNewFiles = () =>{
    this.setState({
      openUploadFileModal: true
    })
  }

  printNotNull = () => {
    const { currentCase,  files } = this.state;
    if (currentCase && currentCase.patientData){
      Object.keys(currentCase.patientData)
      for(const item in currentCase.patientData){
        if (currentCase.patientData[item]){
          console.log(`${item}: ${currentCase.patientData[item]}`);
        }
      }
    }
    return "hello"
  }

  handleFileModalClose = () => {
    this.setState({
      openUploadFileModal: false
    })
  }

  handleUploadFiles = () => {
    const { currentCase,  files } = this.state;
    const current_time = new Date();
    const current_timestamp = current_time.getTime();
    let newCaseData = {
      caseId: currentCase.caseData.case_id,
      last_updated_by_id: this.props.userdata.id,
      is_read: true,
      updated_at: current_timestamp,
      questions: currentCase.caseData.questions,
      is_deleted: false
    };

    if(this.props.userdata.user_type === 'PCP'){
      let caseData = currentCase.caseData; 
      let newFiles = [];  
      if(caseData && caseData.id){
        let caseOldFiles = caseData.existedFiles;
        if (files && files.length >= 1 && caseOldFiles && caseOldFiles.length >=1) {
          files.map((file, i) => (
            caseOldFiles.push(file)
          ));
        }else{
          caseOldFiles = files;
        }
        newCaseData['files'] = caseOldFiles
      }
    }
    
    this.props.modifyCase(newCaseData);

    const currentCaseCondition = {
      case_id: currentCase.caseData.case_id,
      currentUserRole: this.props.userdata.user_type
    }
    setTimeout(() => {
      this.props.setCurrentCase(currentCaseCondition);
    }, 100)
    

    this.setState({
      openUploadFileModal: false,
      files: [],
    })
  }

  uploadFile = async () => {
    this.setState({ isLoading: true });
    const { oldFileList, filePaths } = this.state;
    const promise = await Promise.all(oldFileList.map(async (file) => {
      return await uploadFile(file);
    }));
    promise.map(path => filePaths.push(path));
    axios.post(getServerUrl() + '/api/v1/case/getFileLink', filePaths)
      .then(res => {
        this.setState({
          uploadedLink: res.data.data,
          isLoading: false,
          oldFileList:[]
        })
      })
      .catch(err => {
        this.setState({ isLoading: false })
      })
  }

  removeFile = async fileIndex => {
    const promiseFile = await removeFileFromS3(this.state.files[fileIndex], fileIndex);

    var array = [...this.state.files]; // make a separate copy of the array
    var index = promiseFile

    var urlArray = [...this.state.fileUrls]; // make a separate copy of the array
    var urlIndex = fileIndex;
    if (index !== -1) {
      array.splice(index, 1);
      this.setState({ files: array });

      urlArray.splice(fileIndex, 1);
      this.setState({ fileUrls: urlArray });   
    }
  }

  pdfNext = () => {
    this.setState({pageNumber : this.state.pageNumber + 1});
  }
  pdfPrev = () => {
    this.setState({pageNumber : this.state.pageNumber - 1});
  }

  fileDisplay = (file, fileUrl, fileName) => {
    const pageNumber = 1;
    const fileType = file.type.substring(0, file.type.indexOf("/"));
    let fileExt = file.type.split('/').pop();
    let fileDisplay;

    if (fileType === 'image') {
      fileDisplay = <img src={fileUrl} alt={fileName} style={{ height: '100%', overflow: 'hidden', display: 'block', width: 'auto', justifyContent: "center", alignItems: "center", }} />
    }
    else if (fileType === 'application' && fileExt === 'pdf') {
      fileDisplay = <div><Document
        file={fileUrl}
        alt={fileName}
        style={{ width: "100%", height:"100%", overflow:'scroll', width: "100%", justifyContent: "center", alignItems: "center",}}>
        <Page pageNumber={pageNumber} />
      </Document></div>
    }
    else if (fileType === 'application' && fileExt !== 'pdf') {
      fileDisplay = <h6 style={{ marginLeft: "5%", marginRight: "5%",marginTop: "30%", alignItems: "center", color: "grey", variant: "subtitle2", display:"block",  }}> {fileExt} file </h6>
    }
    else if (fileType === 'video') {
      fileDisplay = <video src={fileUrl} alt={fileName} style={{ width: "100%", height: "100%" }} />                   //Edit this if required
    }
    else {
      fileDisplay = <h6 style={{ marginLeft: "5%", marginRight: "5%",marginTop: "30%", alignItems: "center", color: "grey", variant: "subtitle2", display:"block",  }}> {fileExt} file </h6>
    }
    return fileDisplay;
  }

  render() {
    const { classes, fileObj, expanded, fileIndex } = this.props;
    const { currentCase, openUploadFileModal, uploadedLink, files, fileUrls, fileName, displayModal, pageNumber, numPages, activeStep } = this.state;
    
    let filesUploaded;
    if (files && files.length >= 1) {
      filesUploaded = files.map((file, i) => (
          <img src={this.state.fileUrls[i]} alt={this.state.fileName[i]} style={{ width: "auto", height: "24vh", backgroundColor: "grey" }}  />
      ));
    } else {
      filesUploaded = []
    }

    let uploadButton = '';
    if(this.props.userdata.user_type === 'PCP' && currentCase.caseData && !currentCase.caseData.is_deleted){
      uploadButton = (<div style={{ marginLeft: "auto" }}>
        <Button
            variant="contained"
            color="primary"
            onClick={this.uploadNewFiles}
            className={classes.uploadButton}
          >
            Upload Files
          </Button>
      </div>)
    }
    
    if (expanded) {
      return (
        <Card className={classnames(classes.rightCard, classes.rightCardCollapsed)}>
          <img alt="required" src={DiagnosisIcon} />
        </Card>
      )
    } else {
      return (
        <Card className={classes.rightCard}>
          <div style={{ display: "flex", backgroundColor: "white" }}>
            <Typography className={classes.caseInfoText}>
              Case Information
            </Typography>
            {uploadButton}
          </div>
          
          {/*-----------------Upload case Files ---------------*/}
         <Modal 
          aria-labelledby="simple-modal-title"
          aria-describedby="simple-modal-description"
          open={openUploadFileModal}
          onClose={this.handleClose}
          style={{ display: "flex", justifyContent: "center" }}
          >
            <div className={classes.modalContainer}>
              <Card className={classes.modalCard}>
                <div style={{height: "80%", minHeight: 100 }}>
                  <DropzonePlugin
                    removeFile={this.removeFile}
                    handleFileUpload={this.handleFileUpload}
                    files={this.state.files}
                  />
                </div>

                <Divider />

                <aside style={{ maxHeight: 110, overflow: 'auto', backgroundColor: "white", margin: "10px" }}>
                  {this.state.isLoading && (
                    <CircularProgress
                      style={{ marginTop: 10, marginLeft: 32, alignSelf: 'center' }}
                      size={20}
                      disableShrink
                    />
                  )}
                  <GridList className={classes.gridList} cols={3.5}>
                      {files && files.length >= 1 && (
                        filesUploaded = files.map((file, i) => (
                        <GridListTile key={i}>
                          {this.fileDisplay(file,this.state.fileUrls[i],this.state.fileName[i] )}
                          <GridListTileBar
                          //Add styles to the name(title) to display it smaller for complete name viewing
                            title={file.name.slice(13)}                                 //Slices the initial substring of Epoch time (13 char) from the filename
                            titlePosition="top"
                            classes={{
                              root: classes.titleBar,
                              title: classes.title,
                            }}
                            actionIcon={
                              <IconButton aria-label={`star ${file.title}`}>
                                <HighlightOff style={{ marginLeft: 5, color:"white" }} onClick={this.removeFile.bind(this, i)} className={classes.title}/>
                              </IconButton>
                            }
                          />
                        </GridListTile>
                      ))
                    )}
                  </GridList>
                </aside>

                <Button
                  variant="contained"
                  color="primary"
                  onClick={this.handleUploadFiles}
                  disabled={!Boolean(filesUploaded.length >= 1)}
                >
                  Upload
                </Button>
              </Card>
              <div style={{
                justifyContent: "center",
                display: "flex",
                paddingTop: 5
              }}>
                <Fab
                  variant="round"
                  className={classes.modalCloseButton}
                  onClick={this.handleFileModalClose}>
                  <CloseIcon />
                </Fab>
              </div>
              
            </div>
          </Modal> 

          <div style={{
            justifyContent: "center",
            display: "flex",
            height: "15%",
            alignItems: "center"
          }}>
            {!isEmpty(currentCase) && (
              <Typography className={classes.nameText}>
                <p>Patient: {currentCase.caseData.patient_name}</p>
                {currentCase.patientData.dob && (<p>DOB: {moment(currentCase.patientData.dob).format('MM/DD/YYYY')}</p>)}
              </Typography>

            )}
          </div>

          <div style={{
            justifyContent: "center",
            display: "flex",
            height: "15%",
            alignItems: "center"
          }}>
            {!isEmpty(currentCase) && (
              <Typography className={classes.nameText}>
                <p>{this.printNotNull}</p>
              </Typography>

            )}
          </div>

          {currentCase.caseData !== undefined && currentCase.caseData.files && currentCase.caseData.files.length >= 1 &&
            this.__renderViewer(currentCase.caseData.files)
          }
        </Card>
      )
    }
  }
}

InsideMessageRightCard.propTypes = {
  currentCase: PropTypes.any.isRequired,
  modifyCase: PropTypes.func.isRequired,
  userdata: PropTypes.object.isRequired,
  setCurrentCase: PropTypes.func.isRequired
}

const mapStateToProps = state => ({
  currentCase: state.case.currentCase,
  errors: state.errors,
  userdata: state.auth.userdata
});

export default connect(
  mapStateToProps,
  {modifyCase, setCurrentCase}
)(withStyles(styles)(InsideMessageRightCard));