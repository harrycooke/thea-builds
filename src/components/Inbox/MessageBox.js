import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({

});

export class MessageBox extends Component {
  render() {
    return (
      <div></div>
    );
  }
}

export default withStyles(styles)(MessageBox);