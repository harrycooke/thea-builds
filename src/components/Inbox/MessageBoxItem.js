import React, { Component } from 'react';
import { withStyles } from '@material-ui/styles';
import {
  Grid,
  Card,
  Typography,
  Avatar,
  Checkbox
} from "@material-ui/core"

const styles = theme => ({
  root: {
    padding: theme.spacing(2),
    marginBottom: theme.spacing(2)
  },
  avatarWrap: {
    padding: 8
  }
})

export class MessageBoxItem extends Component {
  render() {
    const { classes } = this.props
    return (
      <Card className={classes.root}>
        <Grid container justify="center" spacing={1}>
          <Grid item xs={2} style={{ padding: 8 }}>
            <Avatar src={require("../../assets/images/alice.png")} />
          </Grid>
          <Grid item xs={7}>
            <Typography variant="h6">Dr. Anita Ramos</Typography>
            <label variant="subtitle1">Re. John Smith</label>
          </Grid>
          <Grid item xs={2}>
            <label>11:30am</label>
          </Grid>
          <Grid item xs={1}>
            <Checkbox color="primary" style={{ padding: 0 }} />
          </Grid>
        </Grid>
        <Typography color="textSecondary" style={{ marginTop: 10 }}>
          Looks like the patient is suffering from macular edema caused by their diabetes. Order OCT to confirm.
        </Typography>
      </Card>
    )
  }
}

export default withStyles(styles)(MessageBoxItem)