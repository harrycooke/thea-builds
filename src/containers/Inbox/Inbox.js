import React, { Fragment } from 'react';
import { Redirect, Link } from 'react-router-dom';
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import MySnackbarContentWrapper from '../../components/SharedComponents/SnackBar';
import {
  Card,
  Grid,
  Typography,
  Button,
  Hidden,
  MenuItem,
  TextField,
  Divider,
  InputBase,
  InputAdornment,
  Snackbar,
} from '@material-ui/core';
import Media from 'react-media';
import SearchIcon from '@material-ui/icons/Search';
import AddIcon from "@material-ui/icons/Add";
import CreateIcon from '@material-ui/icons/Create';

import _ from "../../@lodash";
import { fetchAllCases, setCurrentCase, modifyCase, setSnackbar } from "../../actions/caseAction";
import { fetchAllDrafts, setCurrentDraft, removeDraft } from "../../actions/draftActions";
import MessageList from "../../components/Inbox/MessageList";
import styles from "../../assets/styles/InboxStyles";
import { specialtyList } from "../../utils/constant";
import ConfirmationDialog from "../../components/SharedComponents/ConfirmationDialog"
import { removeFileFromS3 } from '../../utils/s3';
import CircularProgress from '../../components/SharedComponents/CircularProgress';

import { sendAmplitudeData } from '../../utils/amplitude';

const AdapterLink = React.forwardRef((props, ref) => <Link innerRef={ref} {...props} />);

class Inbox extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      redirectToChat: false,
      redirectToAssignConsult: false,
      cases_id: null,
      specialty: null,
      redirectToCreateCase: false,
      readAllCases: [],
      readAllDrafts: [],
      selectedSpecialty: '',
      searchResultsForCases: [],
      searchResultsForDrafts: [],
      confirmDraftDelete: false,
      files: []
    }
    this.removeDraft = this.removeDraft.bind(this)
    this.openDraftDialog = this.openDraftDialog.bind(this)
    this.handleClose = this.handleClose.bind(this)
  }

  handleTrClicked = (id, isDraft, caseData) => {
    if (isDraft) {
      this.setState({
        redirectToCreateCase: true,
        selectedClientID: id
      });
      this.props.setCurrentDraft({ case_id: id })
    } else {
      let newCaseData = {
        caseId: id,
        last_updated_by_id: this.props.userdata.id
      }
      if(this.props.userdata.id === caseData.created_by_id){
        newCaseData['is_read'] = true;
      }else{
        let readByOthers = caseData.is_read_by_other ? caseData.is_read_by_other : [];
        let ifIndex = _.indexOf(readByOthers, this.props.userdata.id);
        if(ifIndex < 0){
          readByOthers.push(this.props.userdata.id);
          newCaseData['is_read_by_other'] = readByOthers;
        }
      }
      this.props.modifyCase(newCaseData);
      this.setState({
        redirectToChat: true,
        selectedClientID: id
      });
    }
  }

  handleSearch = e => {
    e.preventDefault();
    this.setState({
      [e.target.name]: e.target.value
    })
    let selectedSpecialty = e.target.value;
    const { readAllCases, readAllDrafts } = this.state;
    if (selectedSpecialty !== '') {
      let searchResultsForCases = readAllCases;
      let searchResultsForDrafts = readAllDrafts;
      // debugger;
      if(selectedSpecialty != "All Specialties"){                                         //Need to bug-proof here. If the text value to if condition is changed, the condition will break and no value returned.
        searchResultsForCases = _.filter(readAllCases, function (item) {
          return item.caseData.specialty.toLowerCase().indexOf(selectedSpecialty.toLowerCase()) > -1 || item.caseData.patient_name.toLowerCase().indexOf(selectedSpecialty.toLowerCase()) > -1;
        });
        searchResultsForDrafts = _.filter(readAllDrafts, function (item) {
          return item.draftData.specialty.toLowerCase().indexOf(selectedSpecialty.toLowerCase()) > -1 || item.draftData.patient_name.toLowerCase().indexOf(selectedSpecialty.toLowerCase()) > -1;
        });  
      }
      this.setState({ searchResultsForCases, searchResultsForDrafts });
    } else {
    // else if (selectedSpecialty == '') {
    //   this.setState({
    //     searchResultsForCases: readAllCases,
    //     searchResultsForDrafts: readAllDrafts
    //   });
    // }
    //   else {
      this.setState({   
        searchResultsForCases: readAllCases,
        searchResultsForDrafts: readAllDrafts
      });
    }
  }

  componentDidMount() {
    const currentUserData = {
      currentUserId: this.props.userdata.id,
      currentUserRole: this.props.userdata.user_type,
      isDeleted: false,
      isCompleted: false
    }
    sendAmplitudeData('Open Inbox page', currentUserData)
    setTimeout(() => {
      this.props.fetchAllCases(currentUserData);
      this.props.userdata.user_type === 'PCP' && this.props.fetchAllDrafts(currentUserData);
    }, 500)
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    const { readAllCases, readAllDrafts, errors } = nextProps;
    if (errors) {
      this.setState({
        errors: errors
      });
    }
    this.setState({
      readAllCases,
      readAllDrafts,
      searchResultsForCases: readAllCases,
      searchResultsForDrafts: readAllDrafts
    })
    if ((nextProps.isDraftDeleted !== this.props.isDraftDeleted) && nextProps.isDraftDeleted) {
      const currentUserData = {
        currentUserId: this.props.userdata.id,
        currentUserRole: this.props.userdata.user_type,
        isDeleted: false,
        isCompleted: false
      }
      setTimeout(() => {
        this.props.fetchAllCases(currentUserData);
        this.props.userdata.user_type === 'PCP' && this.props.fetchAllDrafts(currentUserData);
      }, 500)
    }
  }
  removeDraft() {
    this.setState({ confirmDraftDelete: false })
    const { filesDraft, draftId } = this.state;
    if (filesDraft && filesDraft.length) {
      filesDraft.map(async (fileObj, index) => {
        if (fileObj && fileObj.name) {
          return await removeFileFromS3(fileObj, index);
        }
      })
    }
    this.props.removeDraft({ case_id: draftId })

  }
  handleClose() { this.setState({ confirmDraftDelete: false }) }

  handleSnackbarClose = () => { 
    this.props.setSnackbar(false);
  }

  openDraftDialog(draftData) {
    this.setState({ confirmDraftDelete: true, draftId: draftData.case_id, filesDraft: draftData.files && draftData.files.length ? draftData.files : [] })
  }

  handleAssignConsult = (id, specialty) => {
    if(id){
      this.setState({redirectToAssignConsult: true, cases_id: id, specialty: specialty});
    }
  }

  render() {
    const { classes, userdata, readAllDrafts } = this.props;
    const {
      redirectToChat,
      redirectToCreateCase,
      selectedClientID,
      selectedSpecialty,
      searchResultsForCases,
      searchResultsForDrafts,
      redirectToAssignConsult,
      cases_id,
      specialty
    } = this.state;
    if (redirectToChat) {
      return (<Redirect to={{
        pathname: `/insidemessage/room_${selectedClientID}`,
        state: { from: 'cases' }
      }} />)
    }
    if (redirectToCreateCase) {
      return (<Redirect to={{
        pathname: `/edit-new-case-${selectedClientID}`,
      }} />)
    }
    if(redirectToAssignConsult){
      return (<Redirect to={{
        pathname: `/assign-consult/case_${cases_id}/specialty_${specialty}`,
      }} />)
    }

    let rows = [...searchResultsForDrafts, ...searchResultsForCases];

    return (
      <div className={classes.container}>
        <CircularProgress isLoading={this.props.draftLoader} />
        <CircularProgress isLoading={this.props.caseLoader} />
        {/* 
        <div>
          <Media queries={{
            small: "(max-width: 599px)",
            medium: "(min-width: 600px) and (max-width: 1199px)",
            large: "(min-width: 1200px)"
          }}>
            {matches => (
              <Fragment>
                {matches.small && <p>I am small!</p> && <style>  </style>}
                {matches.medium && <p>I am medium!</p>}
                {matches.large && <p>I am large!</p>}
              </Fragment>
            )}
          </Media>
        </div> */}

        {/* <Hidden lgUp>
          <Grid container spacing={4} style={{ paddingBottom: 20 }}>
            <Grid item xs={5}
              style={{
                flexDirection: "row",
                display: "flex",
                justifyContent: "flex-start"
              }}
            >
              <Button
                color="primary"
                component={AdapterLink}
                to="/inbox/cases"
                variant="contained"
                style={{
                  fontSize: 16,
                  width: 130,
                  // margin: 5
                }}>
                Open Cases
              </Button>
              <Button
                color="default"
                component={AdapterLink}
                to="/inbox/closed_cases"
                variant="contained"
                style={{
                  fontSize: 16,
                  backgroundColor: "#ffffff",
                  width: 130,
                  marginLeft: 20,
                }}>
                Closed Cases
              </Button>
              {readAllDrafts && readAllDrafts.length ?
                <Button
                  color="default"
                  component={AdapterLink}
                  to="/inbox/drafts"
                  variant="contained"
                  style={{
                    fontSize: 16,
                    backgroundColor: "#ffffff",
                    width: 80,
                    marginLeft: 32,
                  }}>
                  Drafts
              </Button> : null}
            </Grid>
            <Grid item xs={7}
              style={{
                display: "flex",
                justifyContent: "flex-end"
              }}
            >
              {userdata.user_type === "PCP" && (
                <Button
                  color="primary"
                  component={AdapterLink}
                  to="/create-new-case"
                  variant="contained"
                  style={{
                    fontSize: 16,
                    marginLeft: 32,
                    height: 45
                  }}>
                  Create a new Case
              </Button>)}
            </Grid>
          </Grid>
        </Hidden> */}
        <Grid container justify="flex-start" spacing={0} className={classes.cardContainer} >
          <Grid item xs={12}>
            <Card style={{
              backgroundColor: "#ffffff",
              padding: 20,
              overflow: "auto",
              height: "100%"
            }}>
              <Grid  container justify="space-between" >
                <Grid style={{ paddingTop: 6, paddingLeft: 15, paddingRight: 15, paddingBottom: 20,
                  display: "flex",
                  flexDirection: "row",
                  justifyContent: "flex-start"}}>
                  <Typography
                    color="textSecondary"
                    variant="h5"
                    fontWeight="100"
                  >
                    Open Cases
                  </Typography>
                  <Divider />
                </Grid>
                <Grid item xs={12}
                  sm={12} 
                  md={4} 
                  lg={4} 
                  xl={4} 
                  style={{
                  display: "flex",
                  flexDirection: "row",
                  justifyContent: "flex-start"
                }}>
                  <TextField
                    fullWidth
                    select
                    id="outlined-adornment-password"
                    className={classes.searchBar}
                    variant="outlined"
                    type={'text'}
                    size="small"
                    label="Filter by Specialty"
                    //label="Specialty type"
                    name={"selectedSpecialty"}
                    value={selectedSpecialty}
                    onChange={this.handleSearch}
                    style={{
                      display: userdata.user_type === "Admin" || userdata.user_type === "PCP"
                        ? "flex"
                        : "none"
                    }}
                  >
                    {specialtyList.map(specialty => (
                      <MenuItem key={specialty.value} value={specialty.value}>{specialty.label}</MenuItem>
                    ))}
                  </TextField>
                </Grid>
                <Grid item xs={12}
                  sm={12} 
                  md={4} 
                  lg={4} 
                  xl={4}   
                  style={{
                  display: "flex",
                  flexDirection: "row",
                  justifyContent: "flex-start"
                }}> 
                    <TextField
                      fullWidth
                      variant="outlined"
                      placeholder="Search"
                      size="small"
                      className={classes.searchBar}
                      InputProps={{
                       startAdornment: (
                         <InputAdornment position="start">
                           <SearchIcon  color="action"/>
                         </InputAdornment>
                       ),
                      }}
                      onChange={this.handleSearch}
                    />
                </Grid>
              </Grid>
              <MessageList
                openDraftDialog={this.openDraftDialog}
                rows={rows}
                isTrClicked={this.handleTrClicked}
                readAllCases={searchResultsForCases}
                readAllDrafts={searchResultsForDrafts}
                assignConsult={this.handleAssignConsult}
              />
            </Card>
            
          </Grid>
        </Grid>
        <ConfirmationDialog
          message="Are you sure you want to delete this draft?"
          title="Delete Draft"
          okOperationDialog={this.removeDraft}
          handleClose={this.handleClose}
          dialogState={this.state.confirmDraftDelete} />

          <Snackbar
            anchorOrigin={{
              vertical: 'bottom',
              horizontal: 'left',
            }}
            open={this.props.showSnackbar}
            autoHideDuration={3000}
            onClose={this.handleSnackbarClose}
          >
            <MySnackbarContentWrapper
              onClose={this.handleSnackbarClose}
              variant="success"
              message="Case sent successfully."
            />
          </Snackbar>
      </div>
    );
  }
}

Inbox.propTypes = {
  userdata: PropTypes.object.isRequired,
  fetchAllCases: PropTypes.func.isRequired,
  setCurrentCase: PropTypes.func.isRequired,
  modifyCase: PropTypes.func.isRequired,
  fetchAllDrafts: PropTypes.func.isRequired,
  setCurrentDraft: PropTypes.func.isRequired,
  readAllCases: PropTypes.array.isRequired
}

const mapStateToProps = state => ({
  userdata: state.auth.userdata,
  readAllCases: state.case.readAllCases,
  readAllDrafts: state.draft.readAllDrafts,
  isDraftDeleted: state.draft.isDraftDeleted,
  draftLoader: state.draft.draftLoader,
  caseLoader: state.case.caseLoader,
  showSnackbar: state.case.showSnackbar
})

export default connect(
  mapStateToProps,
  { fetchAllCases, setCurrentCase, modifyCase, fetchAllDrafts, setCurrentDraft, removeDraft, setSnackbar }
)(withStyles(styles)(Inbox))