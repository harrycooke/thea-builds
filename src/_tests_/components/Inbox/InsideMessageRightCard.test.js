import React from 'react';
import ReactDOM from'react-dom';
import TestRenderer from 'react-test-renderer'; // ES6

import {InsideMessageRightCard} from '../../../components/Inbox/InsideMessageRightCard';
import sinon from 'sinon';
import { mount } from 'enzyme';

describe('InsideMessageRightCard', () => {
    beforeAll = () => {
        this.state = {
            currentCase: '',
            openUploadFileModal:false,
            files: [],
            isLoading: false
          }
          this.removeFile = this.removeFile.bind(this);
    };
    
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(InsideMessageRightCard, div);
    });

    it('renders React components, traversing a ReactElement tree and represents as JSON object', () => {
      const renderer = TestRenderer.create(InsideMessageRightCard);
      renderer.toJSON();
    });

    describe('InsideMessageRightCard', () => {
      it('should render correctly in "debug" mode', () => {
        const component = (InsideMessageRightCard);
        expect(component).toMatchSnapshot();
      });
    });

    // it('calls funcName', () => {
    //     sinon.spy(InsideMessageRightCard.prototype, 'funcName');
    //     const wrapper = mount(<InsideMessageRightCard />);
    //     expect(InsideMessageRightCard.prototype.componentDidMount).to.have.property('callCount', 1);
    //   });
    
    //   it('allows us to set props', () => {
    //     const wrapper = mount(<InsideMessageRightCard bar="baz" />);
    //     expect(wrapper.props().bar).to.equal('baz');
    //     wrapper.setProps({ bar: 'foo' });
    //     expect(wrapper.props().bar).to.equal('foo');
    //   });
    
    //   it('simulates click events', () => {
    //     const onButtonClick = sinon.spy();
    //     const wrapper = mount((
    //       <InsideMessageRightCard onButtonClick={onButtonClick} />
    //     ));
    //     wrapper.find('button').simulate('click');
    //     expect(onButtonClick).to.have.property('callCount', 1);
    //   });
});