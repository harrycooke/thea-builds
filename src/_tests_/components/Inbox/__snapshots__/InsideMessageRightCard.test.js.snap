// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsideMessageRightCard InsideMessageRightCard should render correctly in "debug" mode 1`] = `[Function]`;
